### PR TITLE
Bill/aux bms race changes

### PR DESCRIPTION
**Pull request by bill-luu**
I intend to keep this PR open until the end of scrutineering. Please help me out to check over to see if this change makes sense, and if it will change any behavior in the setContactorsTask.

See commit messages for more description of each change, the second commit has all the non-generated code change.

### CODE REVIEW

File: EpsilonAuxBMS/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h [hunk 1]
```diff
@@ -16,7 +18,7 @@
   ******************************************************************************
   * @attention
   *
-  * <h2><center>&copy; COPYRIGHT(c) 2017 STMicroelectronics</center></h2>
+  * <h2><center>&copy; COPYRIGHT(c) 2016 STMicroelectronics</center></h2>
```
Comment (Reddragoneer1) [inline EpsilonAuxBMS/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h]: Was this changed automatically?

Comment (bill-luu) [reply EpsilonAuxBMS/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f4xx.h]: Yes

File: EpsilonAuxBMS/Inc/UpdateChargeAllowanceTask.h [hunk 2]
```diff
@@ -8,3 +8,5 @@
 // and discharge contactor is min cell voltage is too low
 // arg: osMutexId* auxStatusMutex
 void updateChargeAllowanceTask(void const* arg);
+
+int canHaveChargeContactorDisengaged();
```
Comment (badeniran) [inline EpsilonAuxBMS/Inc/UpdateChargeAllowanceTask.h]: Can you put a quick description of what the function does above the prototype?

File: EpsilonAuxBMS/Src/main.c [hunk 3]
```diff
@@ -507,13 +504,19 @@ void HAL_CAN_RxCpltCallback(CAN_HandleTypeDef* hcan)
         // Voltages are 2 bytes each, and memory is stored in little endian format
         orionStatus.minCellVoltage = (uint16_t)msg->Data[0] | msg->Data[1] << 8; // Min Cell voltage
         orionStatus.maxCellVoltage = (uint16_t)msg->Data[3] | msg->Data[4] << 8; // Max Cell Voltage
+
         orionStatus.canMsgReceived = 1;
     }
-    else if (msg->StdId == DRIVERS_INPUTS_STDID && msg->DLC == 4)
+    else if (msg->StdId == ORION_TEMP_INFO_STDID && msg->DLC == 8)
+    {
+        orionStatus.highTemperature = msg->Data[0];
```
Comment (badeniran) [inline EpsilonAuxBMS/Src/main.c]: I think there should be a `orionStatus.canMsgReceived = 1` in this else if block

File: EpsilonAuxBMS/Src/main.c [hunk 4]
```diff
@@ -507,13 +504,19 @@ void HAL_CAN_RxCpltCallback(CAN_HandleTypeDef* hcan)
...
+    }
+    else if (msg->StdId == ORION_PACK_INFO_STDID && msg->DLC == 8)
     {
-        driversInput.aux = msg->Data[3] >> DRIVERS_INPUTS_AUX_BIT_POSITION & 1;
-        driversInput.forward = msg->Data[3] >> DRIVERS_INPUTS_FORWARD_BIT_POSITION & 1;
-        driversInput.reverse = msg->Data[3] >> DRIVERS_INPUTS_REVERSE_BIT_POSITION & 1;
+        short packCurrentInt =  // Units 0.1 A
+            (msg->Data[0] << 0) |
+            (msg->Data[1] << 8);
+        orionStatus.packCurrent = (float)packCurrentInt / 10.0f;
```
Comment (badeniran) [inline EpsilonAuxBMS/Src/main.c]: Same thing here as the above comment

File: Documentation/AuxBMS.adoc [hunk 5]
```diff
@@ -70,17 +70,33 @@ Note: At any point, the closing of contactors can be stopped and blocked due to
...
-  - If Discharge Enable goes low, then the discharge contactor is opened and the high voltage is disabled.
+* If both of the charge AND discharge relays are opened, the car will trip (open all contactors, disable high voltage and strobe BMS).
+* If Charge Enable goes low, then the charge contactor is opened.
+    ** Orion will tell the charge contactor to open when:
+    *** The highest cell temperature is greater than 44&#176;C - the car should not be charging to prevent damage to the battery.
 
-.The maximum and minimum cell voltages are acted upon - regardless of whether precharge has been completed or not - in the following way:
-* If the maximum voltage is greater than the maximum allowable cell voltage, then the charge contactor is opened.
-* If the minimum voltage is less than the minimum allowable cell voltage, then the discharge contactor is opened and the high voltage is disabled.
+    ** Aux BMS will tell the charge enable to open when:
```
Comment (badeniran) [inline Documentation/AuxBMS.adoc]: should be charge contactor

File: Documentation/AuxBMS.adoc [hunk 6]
```diff
@@ -70,17 +70,33 @@ Note: At any point, the closing of contactors can be stopped and blocked due to
...
+     *** The Orion reports that the high cell voltage is greater than 4.20 V (Battery Maximum) 
+     *** The Orion reports that the highest cell temperature is greater than 44&#176;C, but Orion is reporting a negative pack current, i.e. The car is still charging.
+    *** The Orion reports that the charging current is more than 47 A.
+
+
+* If Discharge Enable goes low, then the discharge contactor is opened and the high voltage is disabled.
+    ** Orion will tell the discharge contactor to open and disable high voltage when:
+    *** The highest cell temperature is greater than 59&#176;C. The car should not be discharging to prevent damage to the battery.
+
+    ** Aux BMS will tell the discharge enable to open and disable high voltage when:
```
Comment (badeniran) [inline Documentation/AuxBMS.adoc]: discharge contactor

**Pull request merged**